reconcile/utils/saasherder.py

```diff
@@ -795,10 +795,22 @@ def populate_desired_state_saas_file(self, spec, ri):
             ri.register_error()
             return
         # filter resources
-        resources = [resource for resource in resources
-                     if isinstance(resource, dict)
-                     and resource.get('kind') in managed_resource_types]
+        rs = []
+        for r in resources:
+            if isinstance(r, dict):
+                kind = r.get('kind')
+                if kind in managed_resource_types:
+                    resources.append(r)
```

Comment:
rs?

Reply:
Totally. Unit testing this one was very hard but I should have tried. :(